llvm/lib/Target/Sparc/SparcISelLowering.cpp

```diff
@@ -3605,6 +3605,12 @@ bool SparcTargetLowering::useLoadStackGuardNode(const Module &M) const {
   return true;
 }
 
+bool SparcTargetLowering::isFNegFree(EVT VT) const {
+  if (Subtarget->isVIS3())
+    return VT == MVT::f32 || VT == MVT::f64;
+  return false;
+}
```

Comment:
This ideally would be completely separate patch from the patterns. This enables independent combines